src/struct_utils.cc

```diff
@@ -105,8 +109,29 @@ StructUtils::FindResult StructUtils::GetStringList(
 
 StructUtils::FindResult StructUtils::GetValue(
     const std::string& nested_names, const google::protobuf::Value*& found) {
-  const std::vector<absl::string_view> name_vector =
-      absl::StrSplit(nested_names, '.');
+  std::vector<std::string> name_vector;
+
+  // Handle namespaced custom claims that by convention start with a URL - as
+  // most URLs contain dots, we extract the URL up to the FQDN and then split
+  // only the URL path by dots.
+  //
+  // e.g. for nested_names "https://example.com/claims.nested.key", name_vector
+  // would be {"https://example.com/claims", "nested", "key"}
+  if (absl::StartsWith(nested_names, "http://") ||
+      absl::StartsWith(nested_names, "https://")) {
+    const std::vector<std::string> url_components =
```

Comment:
I'm wary of doing this URL parsing manually with string manipulation instead of using an actual URL parser.

There's many things that can go wrong. At the very least, we should at least unescape the URL as `%2F` is the same as `/`. 

https://github.com/abseil/abseil-cpp/issues/158#issuecomment-1034695066

Reply:
While that's not wrong, I am thinking: Is there ever a situation where the contents of the token could potentially be URLencoded after having been b64-decoded from the `Authorization` header? Other than some user error maybe. And in that case, don't we want to fail because the token is not properly formatted?

Or is the assumption that the token is always obtained from said header wrong, for this use case, and we could actually run into this issue AND still want to process it as valid?